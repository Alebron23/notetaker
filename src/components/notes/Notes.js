import React, { Component } from 'react';
import { Form, FormControl, Button } from 'react-bootstrap';
import ModalInstance from './ModalInstance';
import SuccessAlert from './SuccessAlert';
import './notes.css';

class Notes extends Component {

    displayNotes() {
        return this.props.displayedNotes.map((notes, index) => 
            <div className="space" key={index}>
                <div className="zebra" key={index}> 
                    <Form inline onSubmit={this.props.openSaveModal} name={index}>
                        <FormControl type="text" name={index} onChange={this.props.handleChange} value={notes} /> 
                        <Button className="button" name={index} onClick={this.props.openDeleteModal}>X</Button>
                    </Form>
                </div>
            </div>
        )  
    } 

    displayAlert(){
        if(this.props.showAlert === true){
            var props = this.props;
            
            //Have a timeout so the alert goes away after 3 seconds
            setTimeout(function(){props.hideAlert()}, 3000, props);
            return(<SuccessAlert />)
        }
    }

    render(){
        return(
            <div className="container"> 
                <h3>Notes</h3>
                {
                    this.displayAlert()
                }
                {   
                   this.displayNotes()
                } 

                {/* These only show when the delete button is pressed or the enter key is pressed */}
                <ModalInstance handleUpdate = {this.props.handleDelete}
                               title        = {'Are you sure you want to delete this note?'} 
                               hide         = {this.props.closeDeleteModal}
                               show         = {this.props.showDeleteModal} />

                <ModalInstance handleUpdate = {this.props.handleSaveChange} 
                               title        = {'Are you sure you want to save changes to this modal?'}
                               hide         = {this.props.closeSaveModal} 
                               show         = {this.props.showSaveModal} />
                                
            </div>
        );
    }
}

export default Notes;