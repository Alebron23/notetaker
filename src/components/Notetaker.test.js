import React from 'react';
import { mount, shallow } from 'enzyme';
import Notetaker from './Notetaker';
import Notes from './notes/Notes.js';

// Begin test with describe block that describes our
// overall test. 
describe('Notetaker', () => {

    // Now setup data for our test. 
    let notetaker = shallow(<Notetaker/>);

    // it blocks takes two parameters to run the test.
    // The description and the function that runs the actual test
    it('renders the App title', () => {
        // Before you do anything, run a conosle.log(debug)
        // to see if you mounted it correctly
        //console.log(notetaker.debug());
        let title='Note To Self';
        expect(notetaker.find('h1').text()).toEqual(title);
    });

    describe('rendeing the imported Components', () => {
        it('renders the Input component', () => {
            expect(notetaker.find('Input').exists()).toBe(true);
        });

        it('renders Notes component', () =>{
            expect(notetaker.find('Notes').exists()).toBe(true);
        })
    });
});

