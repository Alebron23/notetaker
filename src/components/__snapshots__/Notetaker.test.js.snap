// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notaker practice 1`] = `
<div
  className="container"
>
  <h1>
    Note To Self
  </h1>
  <form
    className=""
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      id={undefined}
      name={Array []}
      onChange={[Function]}
      placeholder="Enter Note"
      type="text"
      value=""
    />
    <div
      style={
        Object {
          "height": "5px",
          "width": "5px",
        }
      }
    >
       
    </div>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "purple",
          "color": "white",
        }
      }
      type="button"
    >
      Submit 
    </button>
  </form>
  <hr />
  <div
    className="container"
  >
    <h3>
      Notes
    </h3>
    <div
      className="static-modal"
    />
    <div
      className="static-modal"
    />
  </div>
</div>
`;
